File: OnlineLogViewer/OnlineLogViewer/log/LogCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OnlineLogViewer.log
{
    public class LogConstants
    {
        public const string PROFILE_NAME_DEV = "dev";
        public const string PROFILE_NAME_TEST = "test";
        public const string PROFILE_NAME_PRODUCT = "production";
        public static string PROFILE_NAME_CURRENT = PROFILE_NAME_PRODUCT;

        public static string EXCHANGE_NAME_COMMAND = "dms.fanout." + PROFILE_NAME_CURRENT + ".log.command";
        public static string EXCHANGE_NAME_COMMAND_RESPONSE = "dms.fanout." + PROFILE_NAME_CURRENT + ".log.command_response";

        public static string EXCHANGE_NAME_OUTPUT = "dms.fanout." + PROFILE_NAME_CURRENT + ".log.output";

        public static void initProfile()
        {
            EXCHANGE_NAME_COMMAND = "dms.fanout." + PROFILE_NAME_CURRENT + ".log.command";
            EXCHANGE_NAME_COMMAND_RESPONSE = "dms.fanout." + PROFILE_NAME_CURRENT + ".log.command_response";
            EXCHANGE_NAME_OUTPUT = "dms.fanout." + PROFILE_NAME_CURRENT + ".log.output";
        }
    }
    public class Logger
    {
        public String contextName { get; set; }
        public string loggerName { get; set; }
        public string loggerLevel { get; set; }
        public string setLoggerLevel { get; set; }
        public bool configured { get; set; }
    }

    public class LogCommandResponse
    {        
        public string command { get; set; }
        public string contextName { get; set; }
        public string mesasge;
        public List<Logger> probeLoggers { get; set; }
    }
    public class ProbeServiceResponse
    {
        public string command { get; set; }
        public string contextName { get; set; }        
    }
    public class LogCommand
    {
        public const string COMMAND_PROBE = "PROBE";
        public const string COMMAND_PROBE2 = "PROBE2";
        public const string COMMAND_PROBE_RESPONSE = "PROBE_RESPONSE";
        public const string COMMAND_CONFIG_LOGGER = "CONFIG_LOGGER";
        public const string COMMAND_RESET_LOGGER_CONFIG = "RESET_LOGGER_CONFIG";
        public const string COMMAND_RESET_ALL_LOGGER_CONFIG = "RESET_ALL_LOGGER_CONFIG";

        public const string COMMAND_LOG_OUTPUT = "LOG_OUTPUT";

        public const string COMMAND_PROBE_SERVICE = "PROBE_SERVICE";
        public const string COMMAND_PROBE_SERVICE_RESPONSE = "PROBE_SERVICE_RESPONSE";

        public string command { get; set; }
        public String contextName { get; set; }
        public LogCommand(string command)
        {
            this.command = command;
        }
    }

    public class ProbeLoggerCommand : LogCommand
    {
        public ProbeLoggerCommand() : base(COMMAND_PROBE)
        {

        }        
    }
    public class ProbeLoggerCommand2 : LogCommand
    {
        public String loggerName { get; set; }
        public ProbeLoggerCommand2() : base(COMMAND_PROBE2)
        {

        }
    }
    public class ConfigLoggerCommand : LogCommand
    {
        public List<Logger> probeLoggers { get; set; }
        
        public ConfigLoggerCommand() : base(COMMAND_CONFIG_LOGGER)
        {

        }
    }
    public class ResetLoggerConfigCommand : LogCommand
    {
        public List<Logger> probeLoggers { get; set; }

        public ResetLoggerConfigCommand() : base(COMMAND_RESET_LOGGER_CONFIG)
        {

        }
    }

    public class ResetAllLoggerConfigCommand : LogCommand
    {
        public ResetAllLoggerConfigCommand() : base(COMMAND_RESET_ALL_LOGGER_CONFIG)
        {

        }
    }

    public class ProbeServiceCommand : LogCommand
    {
        public ProbeServiceCommand() : base(COMMAND_PROBE_SERVICE)
        {

        }
    }
}

File: log-open/src/main/java/com/yyn/base/log/params/LogCommand.java
/**
 * 
 */
package com.yyn.base.log.params;

import java.util.List;

/**
 * @author dandan
 *	日志相关的命令
 */
public class LogCommand {

	public static class LogCommandResponse {
		private String command;

		public LogCommandResponse(String command) {
			super();
			this.command = command;
		}

		public String getCommand() {
			return command;
		}

		public void setCommand(String command) {
			this.command = command;
		}
	}

	public static class ProbeResponse extends LogCommandResponse {

		public static class ProbeLogger {
			private String loggerName;
			private String loggerLevel;
			private boolean configured;

			public String getLoggerName() {
				return loggerName;
			}

			public void setLoggerName(String loggerName) {
				this.loggerName = loggerName;
			}

			public String getLoggerLevel() {
				return loggerLevel;
			}

			public void setLoggerLevel(String loggerLevel) {
				this.loggerLevel = loggerLevel;
			}

			public boolean isConfigured() {
				return configured;
			}

			public void setConfigured(boolean configured) {
				this.configured = configured;
			}

		}

		private String contextName;
		private List<ProbeLogger> probeLoggers;

		public ProbeResponse(String contextName, List<ProbeLogger> probeLogger) {
			super(COMMAND_PROBE_RESPONSE);
			this.contextName = contextName;
			this.probeLoggers = probeLogger;
		}

		public String getContextName() {
			return contextName;
		}

		public void setContextName(String contextName) {
			this.contextName = contextName;
		}

		public List<ProbeLogger> getProbeLoggers() {
			return probeLoggers;
		}

		public void setProbeLoggers(List<ProbeLogger> probeLoggers) {
			this.probeLoggers = probeLoggers;
		}

	}

	public static class LogOutput extends LogCommandResponse {
		private String contextName;
		private String mesasge;

		public String getContextName() {
			return contextName;
		}

		public void setContextName(String contextName) {
			this.contextName = contextName;
		}

		public String getMesasge() {
			return mesasge;
		}

		public void setMesasge(String mesasge) {
			this.mesasge = mesasge;
		}

		public LogOutput(String contextName, String mesasge) {
			super(COMMAND_LOG_OUTPUT);
			this.contextName = contextName;
			this.mesasge = mesasge;
		}

	}

	public static class ProbeServiceResponse extends LogCommandResponse {
		private String contextName;

		public ProbeServiceResponse(String contextName) {
			super(COMMAND_PROBE_SERVICE_RESPONSE);
			this.contextName = contextName;
		}

		public String getContextName() {
			return contextName;
		}

		public void setContextName(String contextName) {
			this.contextName = contextName;
		}
	}

	public static class LoggerConfig {
		private String contextName;
		private String loggerName;
		private String loggerLevel;

		public String getContextName() {
			return contextName;
		}

		public void setContextName(String contextName) {
			this.contextName = contextName;
		}

		public String getLoggerName() {
			return loggerName;
		}

		public void setLoggerName(String loggerName) {
			this.loggerName = loggerName;
		}

		public String getLoggerLevel() {
			return loggerLevel;
		}

		public void setLoggerLevel(String loggerLevel) {
			this.loggerLevel = loggerLevel;
		}
	}

	public final static String COMMAND_PROBE = "PROBE";
	public final static String COMMAND_PROBE2 = "PROBE2";
	public final static String COMMAND_PROBE_RESPONSE = "PROBE_RESPONSE";
	public final static String COMMAND_SET_LOGGER_LEVEL = "CONFIG_LOGGER";
	public final static String COMMAND_RESET_LOGGER_LEVEL = "RESET_LOGGER_CONFIG";
	public final static String COMMAND_RESET_ALL_LOGGER = "RESET_ALL_LOGGER_CONFIG";

	public final static String COMMAND_LOG_OUTPUT = "LOG_OUTPUT";

	public final static String COMMAND_PROBE_SERVICE = "PROBE_SERVICE";
	public final static String COMMAND_PROBE_SERVICE_RESPONSE = "PROBE_SERVICE_RESPONSE";

	private String command;
	private String contextName;
	private String loggerName;
	private List<LoggerConfig> probeLoggers;

	public String getCommand() {
		return command;
	}

	public void setCommand(String command) {
		this.command = command;
	}

	public String getContextName() {
		return contextName;
	}

	public void setContextName(String contextName) {
		this.contextName = contextName;
	}

	public String getLoggerName() {
		return loggerName;
	}

	public void setLoggerName(String loggerName) {
		this.loggerName = loggerName;
	}

	public List<LoggerConfig> getProbeLoggers() {
		return probeLoggers;
	}

	public void setProbeLoggers(List<LoggerConfig> probeLoggers) {
		this.probeLoggers = probeLoggers;
	}

}
